extensions/positron-r/src/session.ts

```diff
@@ -737,6 +737,21 @@ export class RSession implements positron.LanguageRuntimeSession, vscode.Disposa
 		}
 	}
 
+	public async activateLsp() {
+		// Start LSP for the foreground session only if its been previously stopped
+		if (this._lsp.state === LspState.stopped) {
+			const port = await this.adapterApi!.findAvailablePort([], 25);
+			this._lsp.activate(port, this.context);
+		}
+	}
+
+	public deactivateLsp() {
+		// Stop LSP if it's running
+		if (this._lsp.state === LspState.running) {
+			this._lsp.deactivate(true);
+		}
+	}
```

Comment:
And @lionel- you think ark needs the tower-lsp patch so `this._client!.stop()` over in `lsp.ts` actually ends up stopping the LSP on the ark side?